docs/DeveloperGuide.md

```diff
@@ -2,28 +2,454 @@
+    - `date`: represents the date to be tagged of transaction, stored as LocalDate object
+
+### Adding an entry `add`
+
+- When a command is given to add a statement, the command is first parsed to check whether it follows the format of an
+  add command: `add -DIRECTION DESCRIPTION $VALUE -c CATEGORY -date DD/MM/YYYY` with the use of regex pattern
+    - Details on implementation for parsing and command fields are documented below
+- Commands in the correct format will then be parsed to extract the relevant information, and an `AddCommand` object
+  will be created with the relevant attributes
+- `RainyDay` will then call `execute` method in `Command`, where the transaction will be added into the financial report
+
+The sequence diagram for the implementation of add is as shown below:
+
+![AddCommandSequenceDiagram.png](images\DeveloperGuide\AddCommandSequenceDiagram.png)
+
+#### Design considerations
+
+Format of add command
+
+- Alternative 1 (current choice): usage of flags
+    - Pros: able to identify the arguments easily, as the combination of the characters are highly unlikely to be used
+      as a description or category name
+    - Cons: not that intuitive from the user's point of view
+- Alternative 2: usage of keywords
+    - Pros: easier to use from the user's point of view
+    - Cons: unable to correctly identify the keyword as the sequence of characters may be a description or category that
+      the user wants to use
+
+### Deleting an entry `delete`
+
+- When a command is given to delete a statement, the command is first parsed to check whether it follows the format of a
+  delete command: `delete INDEX` with the use of regex pattern
+    - `INDEX`: the index number of the statement in the financial report, stored as an int
+- Commands in the correct format will then be parsed to extract index, and a `DeleteCommand` object will
+  be created with the relevant attribute information
+- `RainyDay` will then call `execute` method in `Command`, where the indicated transaction will be deleted from the
+  financial report
+
+The sequence diagram for the implementation of delete is as shown below:
+
+![DeleteCommandSequenceDiagram.png](images\DeveloperGuide\DeleteCommandSequenceDiagram.png)
+
+#### Design considerations
+
+Format of delete command
+
+- Alternative 1 (current choice): usage of index to identify the statement to be deleted
+    - Pros: able to identify the correct statement to be deleted following the index assigned to the statement
+    - Cons: may not be intuitive for the user, as they have to look up the index of the statement before deletion
+- Alternative 2: usage of full statement information to identify the statement to be deleted
+    - Pros: more intuitive for the user
+    - Cons: user needs to know the full details of the statement and needs to type the full statement information for
+      identifying the statement to be deleted
+
+### Implementation of regex and parser
```

Comment:
This section feels a bit out of place as the previous and next sections were about different commands. Would shifting this section be better?